locallib.php

```diff
@@ -58,4 +58,82 @@ function install_and_remove_block() {
             set_config("active_feedback", 0, "block_coursefeedback");
         }
     }
+}
+
+/**
+ * Returns the coursefeedback results for all courses and a given feedback and question
+ * Returns only the results for a specific course if $specificcourseid is provided
+ *
+ * @param int $questionid
+ * @param int $coursefeedbackid
+ * @param int $answerlimit
+ * @param int $showperpage
+ * @param int $page
+ * @return array of objects
+ * @throws \moodle_exception
+ */
+function block_coursefeedback_get_courserankings(
+        $questionid, $coursefeedbackid, $specificcourseid = 0,  $answerlimit = 0, $showperpage = 0, $page = 0) {
+    global $DB;
+    // Get courseids and the amount of answers in this course for the current question.
+    $params = [
+        'feedbackid' => $coursefeedbackid,
+        'feedbackid2' => $coursefeedbackid,
+        'questionid' => $questionid,
+        'questionid2' => $questionid,
+        'answerlimit' => $answerlimit,
+        'specificcourseid' => $specificcourseid,
+        'specificcourseid2' => $specificcourseid
+    ];
+    $sql = "
```

Comment:
Sehr schön, auch die Kommentare im SQL. Love to see it. Ich vertraue dir, dass du die Abfrage ausgiebig getestet hast. :smile:

Reply:
Ja hab ich :)